code/web/src/modules/user/EditProfile.js

```diff
@@ -0,0 +1,313 @@
+// Imports
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { Helmet } from 'react-helmet'
+import { Link } from 'react-router-dom'
+import { withRouter } from 'react-router-dom'
+
+// UI Imports
+import { Grid, GridCell } from '../../ui/grid'
+import Button from '../../ui/button'
+import { H3, H4 }  from '../../ui/typography'
+import { Input, Textarea, Select } from '../../ui/input'
+import { grey, grey2 } from '../../ui/common/colors'
+import { level1 } from '../../ui/common/shadows'
+import { primary } from '../../ui/common/fonts'
+
+// App Imports
+// import admin from '../../../setup/routes/admin'
+import { routeImage } from '../../setup/routes'
+import { renderIf, slug } from '../../setup/helpers'
+import { logout } from './api/actions'
+import { upload, messageShow, messageHide } from '../common/api/actions'
+import { APP_URL } from '../../setup/config/env'
+import { statesList } from './helperData/statesList'
```

Comment:
Really nice job figuring out what you had to import to make this work. I can imagine that being a very tricky task.

Reply:
Final comment